opal/corelib/kernel.rb

```diff
@@ -1006,21 +1006,38 @@ def warn(*strs)
 
   def raise(exception = undefined, string = undefined)
     %x{
-      if (exception == null && #$!) {
+      if (exception == null && #{$! != nil}) {
         throw #$!;
       }
-
       if (exception == null) {
         exception = #{RuntimeError.new};
       }
       else if (exception.$$is_string) {
         exception = #{RuntimeError.new exception};
       }
-      else if (exception.$$is_class) {
+      // using respond_to? and not an undefined check to avoid method_missing matching as true
+      else if (exception.$$is_class && #{exception.respond_to?(:exception)}) {
+        exception = #{exception.exception string};
+      }
+      else if (#{exception.kind_of?(Exception)}) {
+        // exception is fine
+      }
+      else if (exception.$$is_class && (exception === #{Exception} || #{exception < Exception})) {
```

Comment:
@iliabylich - opal's try site is still using 0.8.0 beta1 which was not compliant w/ RubySpecs for the < operator. That issue was fixed in https://github.com/opal/opal/pull/1105/files#r40012949.

If you run a very similar sample to your Opal try link in IRB, you'll get this output:
```ruby
true

NilClass
false
```

code:
```ruby
class ApiException < Exception
end

class Dog
end

puts ApiException < Exception
puts Dog < Exception
puts (Dog < Exception).class
puts Exception < Exception

```